selector/__init__.py

```diff
@@ -383,33 +383,34 @@ def create_pool(self) -> Dict[Site, List[Visit]]:
         scheduling_groups = self.collector.scheduling_groups
         visits = {site: [] for site in self.sites}
 
+        print(self.sites)
         for site in self.sites:
             # Create Visits
-            for idx, group in enumerate(scheduling_groups.values()):
+            for idx, group in enumerate(scheduling_groups[site].values()):
                 obs_idxs = group['idx']
 
-                instruments = [collected_observations[obs].instrument
+                instruments = [collected_observations[site][obs].instrument
                                for obs in obs_idxs]
 
                 wavelengths = set([wav for inst in instruments for wav in inst.wavelength()])
 
-                modes = [[collected_observations[obs].instrument.observation_mode()
+                modes = [[collected_observations[site][obs].instrument.observation_mode()
                           for obs in obs_idxs]]
```

Comment:
I didn't notice this before, but why do we make `modes` into a list containing a single element?
Why the doubly nested list?

Reply:
Why shouldn't we fix it now, then?
I have noticed a lot of code currently goes unused, which is making me a bit nervous about correctness.